crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -1944,26 +1907,24 @@ impl CheckpointService {
             effects_store,
             accumulator,
             checkpoint_output,
-            exit_rcv.clone(),
             notify_aggregator.clone(),
             metrics.clone(),
             max_transactions_per_checkpoint,
             max_checkpoint_size_bytes,
         );
 
-        spawn_monitored_task!(builder.run());
+        tasks.spawn(builder.run());
 
         let aggregator = CheckpointAggregator::new(
             checkpoint_store.clone(),
             epoch_store.clone(),
             notify_aggregator.clone(),
-            exit_rcv,
             certified_checkpoint_output,
             state.clone(),
             metrics.clone(),
         );
 
-        spawn_monitored_task!(aggregator.run());
+        tasks.spawn(aggregator.run());
```

Comment:
we are also losing the spawn_monitored_task! here. should be easy to change the macro to take a joinset.

Reply:
Using monitored_future!() now.